### PR TITLE
Add basic page and file migrations

### PR DESCRIPTION
**Pull request by pfrilling**
## Summary
This adds a new ecms_migration module that will take a google sheet of urls and create basic pages and file/media entities from those URLS.

## Metadata
| Question | Answer |
|----------|--------|
| Did you use a meaningful pull request title? | yes
| Did you apply meaningful labels to the pull request? | yes
| Did you perform a self review first? | yes
| Documentation reflects changes? | yes
| `CHANGELOG` reflects changes? | yes
| Unit/Functional tests reflect changes? | yes (mostly)
| Did you perform browser testing? | yes
| Risk level | Low
| Relevant links | [RIG-106](https://thinkoomph.jira.com/browse/rig-106)

### CODE REVIEW

File: ecms_base/modules/custom/ecms_migration/config/install/migrate_plus.migration.ecms_file_media.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# Migrate a Google spreadsheet of URLs as file entities into the eCMS system.
+id: ecms_file_media
+label: Create media entities from the import files..
```
Comment (ben-hamelin) [inline ecms_base/modules/custom/ecms_migration/config/install/migrate_plus.migration.ecms_file_media.yml:3]: nitpick label punctuation ".."

File: ecms_base/modules/custom/ecms_migration/ecms_migration.module [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+use Drupal\migrate\Plugin\MigrateSourceInterface;
+use Drupal\migrate\Plugin\MigrationInterface;
+use Drupal\migrate\Row;
+use Drupal\Core\Site\Settings;
+
+/**
+ * Implements hook_migrate_MIGRATION_ID_prepare_row() for the ecms_file_redirect migration.
+ */
+function ecms_migration_migrate_ecms_file_redirect_prepare_row(Row $row, MigrateSourceInterface $source, MigrationInterface $migration) {
```
Comment (ben-hamelin) [inline ecms_base/modules/custom/ecms_migration/ecms_migration.module:18]: return type

File: ecms_base/modules/custom/ecms_migration/src/Form/EcmsMigrationConfigForm.php [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+   * {@inheritDoc}
+   */
+  public function getFormId(): string {
+    return 'ecms_migration_settings_form';
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
```
Comment (ben-hamelin) [inline ecms_base/modules/custom/ecms_migration/src/Form/EcmsMigrationConfigForm.php:50]: return type

File: ecms_base/modules/custom/ecms_migration/tests/src/Unit/Form/EcmsMigrationConfigFormTest.php [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+      'google_sheet_id' => 'GOOGLE_ID_456',
+      'css_selector_1' => '#css-selector-1',
+      'css_selector_2' => '#css-selector-2',
+      'css_selector_3' => '#css-selector-3',
+    ],
+    '_core' => 'TEST',
+  ];
+
+  /**
+   * Then expected migration configuration.
```
Comment (ben-hamelin) [inline ecms_base/modules/custom/ecms_migration/tests/src/Unit/Form/EcmsMigrationConfigFormTest.php:48]: nitpick "The"

File: ecms_base/modules/custom/ecms_migration/README.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+you can begin migrating. Click `Execute` next to the migration you would like
+to run, select `import` and click execute. This will pull each url from
+the Google Sheet and will create the appropriate Drupal entities.
+
+## Adding New Migrations
+When adding new migrations to this module you'll need to do a few things:
+
+1. Create the migration and export the configuration to the `config/install` directory.
+2. In the `ecms_migration.settings` configuration, add a new key with the values
+   that need replaced in the migration.
```
Comment (ben-hamelin) [inline ecms_base/modules/custom/ecms_migration/README.md:53]: nitpick "to be replaced"

**Review by ben-hamelin — CHANGES_REQUESTED** [review]
A few nitpicks on comments, a few return types. Overall this is looking great!

**Comment by pfrilling** [comment]
@ben-hamelin, I've addressed all of the QA fixes.

**Comment by ben-hamelin** [comment]
Looks great!
/approved

**Pull request merged**